docs/src/matrix_fields.md

```diff
@@ -98,4 +100,252 @@ column_field2array
+where $M = k * I$, and $k$ is of type `T_k`. This would happen if
+$\frac{\partial f_i}{\partial g_j} = [i=j] * k$. Instead of storing
+each element on the diagonal, less memory can be used by storing a single value that represents a scaling of the identity. This reduces memory usage by a factor of $N_v$.
+
+```julia
+entry = fill(DiagonalMatrixRow(k), space)
+```
+
+can also be represented by
+
+```julia
+entry = DiagonalMatrixRow(k)
+```
+
+or, if `T_k` is a scalar, then
+
+```julia
+entry = I * k
+```
+
+### Implicit Tensor Structure Optimization
+
+The functions that index an entry with an internal key assume the implicit tensor structure optimization is being used
+when all of the following are true for `entry` where `T_k` is the element type of each band, and
+`(internal_key_1, internal_key_2)` is the internal key indexing `entry`.
+
+- the first key in the `internal_key_1` name chain is not a fieldname of `T_k`
+- the first key in the `internal_key_2` name chain is not a fieldname of `T_k`
+- `internal_key_1` and `internal_key_2` are both not empty
+
+For most use cases, `T_k` is a scalar.
+
+If the above conditions are met, the optimization assumes that the user intends the
+entry to have an implicit tensor structure, with the values of type `T_k` representing a scaling of the
+tensor identity. If both the first and second name in the name pair are equivalent, then they index onto the diagonal,
+and the value is returned. Otherwise, they index off the diagonal, and a zero value
+is returned.
+
+This optimization is intended to be used when `T_f = T_g`, and they are both `AxisVectors`
+The notation $f_{n}[i]$ where $0 < n \leq N_v$  refers to the $i$ component of the element
+at the $n$ vertical level of $f$. In the following example, `T_f=T_g=Covariant12Vector`, and
+$b_1 = b_2 = 1$, and
+
+```math
+\frac{\partial f_n[i]}{\partial g_m[j]} = \begin{cases}
+  -0.5, & \text{if } i = j \text{ and }  m = n-1 \text{ or } m = n+1 \\
+  1, & \text{if } i = j \text{ and } m = n \\
+  0, & \text{if } i \neq j \text{ or } m < n -1 \text{ or } m > n +1
+\end{cases}
+```
+
+The non-zero values of each row of `M` are equivalent in this example, but they can also vary in value.
+
+```julia
+∂f_∂g = fill(MatrixFields.TridiagonalMatrixRow(-0.5 * identity_axis2tensor, identity_axis2tensor, -0.5 * identity_axis2tensor), space)
+J = MatrixFields.FieldMatrix((@name(f), @name(g))=> ∂f_∂g)
+```
+
+`∂f_∂g` can be indexed into to get the partial derrivatives of individual components.
+
+
+```@example 2
+J[(@name(f.components.data.:(1)), @name(g.components.data.:(1)))]
+```
+
+```@example 2
+J[(@name(f.components.data.:(2)), @name(g.components.data.:(1)))]
+```
+
+This can be more optimally stored with the implicit tensor structure optimization:
+
+```@example 2
+∂f_∂g = fill(MatrixFields.TridiagonalMatrixRow(-0.5, 1.0, -0.5), space) # hide
+J = MatrixFields.FieldMatrix((@name(f), @name(g))=> ∂f_∂g) # hide
```

Comment:
```suggestion
J = MatrixFields.FieldMatrix((@name(f), @name(g))=> ∂f_∂g) # hide
nothing # hide
```
This is currently rendering the value of `J` instead of hiding it. You need to either return `nothing` from the `@example` block or replace `@example` with `@setup`.